app/assets/javascripts/channels/notification.coffee

```diff
@@ -13,10 +13,23 @@ App.notification = App.cable.subscriptions.create "NotificationChannel",
 
     else if notification.code == 'message'
       match_id = parseInt($('#match').val())
+
       if match_id == notification.message.match_id
         $('#conversation-body').append(notification.message.html_element)
         $("#conversation-container").animate({ scrollTop: $("#conversation-container")[0].scrollHeight}, 1000);
+        $.post({ url: '/matches/mark_as_read', data: { id: notification.message.match_id }});
+
       else
-        # Increment unread messages count
+        if !$('#conversation-icon').hasClass('unread-conversation-badge')
+          $('#conversation-icon').addClass('unread-conversation-badge');
+
+        unread_messages = parseInt($('#matches-table > tbody > tr[data-match-id=' + notification.message.match_id + '] > td > div').html()) || 0
+        $('#matches-table > tbody > tr[data-match-id=' + notification.message.match_id + '] > td > div').show().html(unread_messages + 1);
+
+        global_count = parseInt($('#global-message-count').text().replace(/[^0-9\.]/g, ''), 10)
```

Comment:
instead of this, put the @global_unread_messages_count inside a span, and the just get it with $('#global-message-count span').text()